source/agora/node/Ledger.d

```diff
@@ -1311,6 +1421,34 @@ unittest
     genBlockTransactionsFreeze(1, TxType.Payment, false);
     assert(ledger.getBlockHeight() == 11);
 
+    // Enroll the validator for the second time. (FreezingPeriod / 2 <= X < FreezingPeriod)
+    {
+        genNormalBlockTransactions((FreezingPeriod / 2) - 1 - ledger.getBlockHeight());
+        assert(ledger.getBlockHeight() == FreezingPeriod / 2 - 1);
+
+        auto enrolls = remakeEnrollment(genesis, params);
+        auto findUTXO = utxo_set.getUTXOFinder();
+        foreach (ref enroll; enrolls)
+            assert(enroll_man.pool.add(enroll, findUTXO));
+
+        genNormalBlockTransactions((FreezingPeriod / 2) - ledger.getBlockHeight());
+        assert(ledger.getBlockHeight() == FreezingPeriod / 2);
```

Comment:
Does this code actually re-enroll the validator? Because the test is only testing the block height.

Reply:
When the height of the block is 1007 and 2015, re-enroll the validator.
Re-enrollment is required because validation is checked in the process of making blocks.

Reply:
Ok. Well these tests will probably be rewritten some time in the future, because they have become very complex to follow.